R/download.R

```diff
@@ -233,17 +229,16 @@ renv_download_default_agent_scope_impl <- function(headers, envir = NULL) {
   if (!ok)
     return(FALSE)
 
-  do.call("unlockBinding", list("makeUserAgent", utils))
-  defer(do.call("lockBinding", list("makeUserAgent", utils)), envir = envir)
-
   agent <- makeUserAgent(FALSE)
   all <- c("User-Agent" = agent, headers)
   headertext <- paste0(names(all), ": ", all, "\r\n", collapse = "")
 
-  assign("makeUserAgent", envir = utils, function(format = TRUE) {
+  renv_binding_replace("makeUserAgent", envir = utils, function(format = TRUE) {
```

Comment:
Maybe we can use `renv_scope_var()` (or should we call it `renv_scope_binding()`?)